bindings/python/pymongoarrow/bson.pyi

```diff
@@ -75,6 +77,11 @@ def process_bson_stream(bson_stream, context):
                             builder.append(<bytes>(<uint8_t*>bson_iter_oid(&doc_iter))[:12])
                         else:
                             builder.append_null()
+                    elif ftype == t_string:
+                        if value_t == BSON_TYPE_UTF8:
+                            builder.append(bson_iter_utf8 (&doc_iter, &str_len))
```

Comment:
How does `tobytes` deal with nulls in the value? IIUC bson_iter_utf8 returns a pointer to the C byte string but it looks like we never use str_len.

Reply:
It is a null-terminated string, we have to provide a `str_len` to match the API, but we don't use it.

Reply:
Actually, I see that UTF-8 can have NUL characters embedded, so we do need to handle that case.